### PR TITLE
Rust library for decoding events

### PR DESCRIPTION
**Pull request by erwanncloarec**
Related to: 
* https://github.com/pbstck/intake/issues/321

### CODE REVIEW

**Review by slezier — APPROVED** [review]

File: rust/src/compact.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            return Err(ModelEncodeError::UnexpectedStreamError);
+        }
+    };
+    let mut compression_encoder = async_compression::tokio::write::GzipEncoder::new(record);
+    match compact_events(&mut compression_encoder, events).await {
+        Ok(()) => Ok(compression_encoder.into_inner()),
+        Err(e) => Err(e),
+    }
+
+    // compact_raw_events(model, events, record).await
```
Comment (MichaelBitard) [inline rust/src/compact.rs:22]: dead code

**Review by MichaelBitard — COMMENTED** [review]
I don't see all the tests from the other projects (with the file decoding/recoding)

IMHO that's mandatory to ensure we did not broke anything

**Comment by erwanncloarec** [comment]
We updated the tests so that :
* No snapshot (no need to update snapshot if we add a column)
* Integration tests with a extract/compact/extract chain like in Intake or Darty tests and real pipeline data (ping @MichaelBitard)

File: .github/workflows/ci-rust.yaml [hunk 2]
```diff
@@ -20,10 +20,10 @@ jobs:
           components: rustfmt, clippy
 
       - uses: actions-rs/cargo@v1
-        name: build
+        name: check
```
Comment (MichaelBitard) [inline .github/workflows/ci-rust.yaml:23]: This should use `make lint/check` instead of custom checks right?

**Review by MichaelBitard — APPROVED** [review]

File: .github/workflows/ci-rust.yaml (hunk 2 above)
Comment (erwanncloarec) [reply .github/workflows/ci-rust.yaml:23]: It's not for linting, here it's a cargo check.
But I think you want to add this `make lint/check` in Makefile and change the `fmt` and `clippy` goals in this file right ?

**Review by bgouin — APPROVED** [review]

File: .github/workflows/ci-rust.yaml (hunk 2 above)
Comment (MichaelBitard) [reply .github/workflows/ci-rust.yaml:23]: Yes sorry, that was not really clear :)

**Pull request merged**